components/external-services-mock/internal/destinationfetcher/handler.go

```diff
@@ -0,0 +1,34 @@
+package destinationfetcher
+
+import (
+	"github.com/gorilla/mux"
+	"github.com/kyma-incubator/compass/components/director/pkg/log"
+	"net/http"
+)
+
+type Handler struct{}
+
+func NewHandler() *Handler {
+	return &Handler{}
+}
+
+func (h *Handler) GetSensitiveData(writer http.ResponseWriter, req *http.Request) {
+	ctx := req.Context()
+	destinationName := mux.Vars(req)["name"]
+	log.C(ctx).Infof("Sending sensitive data for destination: %s", destinationName)
+	data, ok := destinationsSensitiveData[destinationName]
+
+	if !ok {
+		http.Error(writer, "Not Found", http.StatusNotFound)
+		return
+	}
+
+	if _, err := writer.Write(data); err != nil {
+		log.C(ctx).Errorf("Failed to respond with error %s", err.Error())
```

Comment:
Change message. Failed to write data: %s
I think now it sounds like it fails to respond with error / as we expect it to fail with this error

Reply:
or better `log.C(ctx).WithError(err).Errorf("Failed to write data")`